sdk/data.go

```diff
@@ -28,7 +28,19 @@ func (i Inventory) SetItem(key string, field string, value interface{}) {
 
 // Event is the data type to represent arbitrary, one-off messages for key
 // activities on a system.
-type Event map[string]interface{}
+type Event struct {
+	Summary  string `json:"summary"`
+	Category string `json:"category,omitempty"`
+}
+
+// NewEvent creates new Event with values passed as arguments.
+func NewEvent(summary, category string) Event {
+	ev := Event{
+		Summary:  summary,
```

Comment:
Which should be the behavior if I pass empty strings as `summary` or `category` arguments?

Reply:
I don't remember if we discuss about it or not. Because in the current code when the category is empty we skip it in the payload. 
```
	"events": [
		{
			"summary": "foo1"
		}
]
```
But to be honest I'm not sure what should be the behaviour. I think that we should fail when `summary` is empty. And ignore `category` when is empty (as we do right now)

Reply:
Ignoring `category` is fine because then internally the agent is assigning a default category: `notifications`. Regarding an empty summary, I agree with you: I consider it an error and we should fail.